src/bin/zmonlyofficeinstall

```diff
@@ -0,0 +1,434 @@
+}
+
+ask() {
+	PROMPT=$1
+	DEFAULT=$2
+	echo ""
+	echo -n "$PROMPT [$DEFAULT] "
+	read response
+	if [ -z $response ]; then
+		response=$DEFAULT
+	fi
+}
+
+askYN() {
+	PROMPT=$1
+	DEFAULT=$2
+	if [ "x$DEFAULT" = "xyes" -o "x$DEFAULT" = "xYes" -o "x$DEFAULT" = "xy" -o "x$DEFAULT" = "xY" ]; then
+		DEFAULT="Y"
+	else
+		DEFAULT="N"
+	fi
+	while [ 1 ]; do
+		ask "$PROMPT" "$DEFAULT"
+		response=$(perl -e "print lc(\"$response\");")
+		if [ -z $response ]; then
+			:
+		else
+			if [ $response = "yes" -o $response = "y" ]; then
+				response="yes"
+				break
+			else
+				if [ $response = "no" -o $response = "n" ]; then
+					response="no"
+					break
+				fi
+			fi
+		fi
+		echo "A Yes/No answer is required"
+	done
+}
+
+configurePackageServer() {
+	echo -e ""
+	response="no"
+	HOSTNAME=`hostname --fqdn`
+	askYN "Use Zimbra's package repository" "Y"
+	if [ $response = "yes" ]; then
+		PACKAGE_SERVER="repo.zimbra.com"
+		response="no"
+		echo $HOSTNAME | egrep -qe 'eng.zimbra.com$|zimbradev.com$' > /dev/null 2>&1
+		if [ $? = 0 ]; then
+			askYN "Use internal development repo" "N"
+			if [ $response = "yes" ]; then
+				PACKAGE_SERVER="repo-dev.eng.zimbra.com"
+			else
+				response="no"
+				askYN "Use internal production mirror" "N"
+				if [ $response = "yes" ]; then
+					PACKAGE_SERVER="repo.eng.zimbra.com"
+				fi
+			fi
+		fi
+	fi
+	echo $PLATFORM | egrep -q "UBUNTU|DEBIAN"
+	if [ $? = 0 ]; then
+		if [ $PLATFORM = "UBUNTU22_64" ]; then
+			repo="jammy"
+		elif [ $PLATFORM = "UBUNTU20_64" ]; then
+			repo="focal"
+		elif [ $PLATFORM = "UBUNTU18_64" ]; then
+			repo="bionic"
+		else
+			print "Aborting, unknown platform: $PLATFORM"
+			exit 1
+		fi
+		echo "Configuring package repository"
+		apt-get install -y apt-transport-https >>$LOGFILE 2>&1
+		if [ $? -ne 0 ]; then
+			 echo "ERROR: Unable to install packages via apt-get"
+			 echo "Please fix system to allow normal package installation before proceeding"
+			 exit 1
+		fi
+		if [ ${ZM_CUR_MAJOR} -eq 10 -a ${ZM_CUR_MINOR} -eq 0 ]; then
+cat > /etc/apt/sources.list.d/zimbra-onlyoffice.list << EOF
+deb     [arch=amd64] https://$PACKAGE_SERVER/apt/onlyoffice $repo zimbra
+EOF
+		else
+cat > /etc/apt/sources.list.d/zimbra-onlyoffice.list << EOF
+deb     [arch=amd64] https://$PACKAGE_SERVER/apt/onlyoffice-${ZM_CUR_MAJOR}${ZM_CUR_MINOR}${ZM_CUR_MICRO} $repo zimbra
+EOF
+		fi
+		apt-get update >>$LOGFILE 2>&1
+		if [ $? -ne 0 ]; then
+			echo "ERROR: Unable to install packages via apt-get"
+			echo "Please fix system to allow normal package installation before proceeding"
+			exit 1
+		fi
+	else
+		if [ $PLATFORM = "RHEL7_64" ]; then
+			repo="rhel7"
+		elif [ $PLATFORM = "RHEL8_64" ]; then
+			repo="rhel8"
+		elif [ $PLATFORM = "RHEL9_64" ]; then
+			repo="rhel9"
+		else
+			print "Aborting, unknown platform: $PLATFORM"
+			exit 1
+		fi
+		echo "Configuring package repository"
+		if [ ${ZM_CUR_MAJOR} -eq 10 -a ${ZM_CUR_MINOR} -eq 0 ]; then
+cat > /etc/yum.repos.d/zimbra-onlyoffice.repo <<EOF
```

Comment:
Heredocuments (<<EOF) are intentionally aligned with surrounding code for readability.